electronics_lib/Passives.py

```diff
@@ -314,6 +297,222 @@ def derated_capacitance(row: Dict[str, Any]) -> Tuple[float, float]:
+  # 0402:                   50v /                0.1uF,     25v / 0.1uF,                      10v / 2.2uF
+  # 0603:                   50v /                0.1uF,     25v /   1uF,     16v / 2.2uF,     10v /  10uF
+  # 0805: 100v / 0.1uF,     50v / 0.1uF (maybe 0.22uF),     25v /  10uF
+  # 1206: 100v / 0.1uF,     50v /                4.7uF,     25v /  10uF,                      10v /  22uF
+  # 1210: 100v / 4.7uF,     50v /                 10uF,                      16v /  22uF,     10v /  47uF
+  # 1812: 100v / 2.2uF,     50v /                  1uF,     25v /  10uF (though small sample size)
+
+  # derating coefficients in terms of %capacitance / V over 3.6
+  # 'Capacitor_SMD:C_0603_1608Metric'  # not supported, should not generate below 1uF
+
+  class SmtCeramicCapacitorGenericPackageSpecs(NamedTuple):
+    max: float
+    derate: float
+    vc_pairs: dict
+
+  PACKAGE_SPECS = {
+    'Capacitor_SMD:C_0402_1105Metric': SmtCeramicCapacitorGenericPackageSpecs(
+      max=1e-7,
+      derate=0,
+      vc_pairs={             50:   1e-7, 25: 1e-7,             10: 2.2e-6},
+    ),
+    'Capacitor_SMD:C_0603_1608Metric': SmtCeramicCapacitorGenericPackageSpecs(
+      max=1.1e-6,
+      derate=0,
+      vc_pairs={             50:   1e-7, 25: 1e-6, 16: 2.2e-6, 10:   1e-5},
+    ),
+    'Capacitor_SMD:C_0805_2012Metric': SmtCeramicCapacitorGenericPackageSpecs(
+      max=11e-6,
+      derate=0.08,
+      vc_pairs={100:   1e-7, 50:   1e-7, 25: 1e-5, },
+    ),
+    'Capacitor_SMD:C_1206_3216Metric': SmtCeramicCapacitorGenericPackageSpecs(
+      max=float('inf'),
+      derate=0.04,
+      vc_pairs={100:   1e-7, 50: 4.7e-6, 25: 1e-5,             10: 2.2e-5},
+    ),
+    'Capacitor_SMD:C_1210_3225Metric': SmtCeramicCapacitorGenericPackageSpecs(
+      max=float('inf'),
+      derate=0,
+      vc_pairs={100: 4.7e-6, 50:   1e-5,           16: 2.2e-5, 10: 4.7e-5},
+    ),
+    'Capacitor_SMD:C_1812_4532Metric': SmtCeramicCapacitorGenericPackageSpecs(
+      max=float('inf'),
+      derate=0,
+      vc_pairs={100: 2.2e-6, 50:   1e-6, 25: 1e-5, },
+    ),
+  }
+
+  def select_capacitor_no_prod_table(self, capacitance: RangeVal, voltage: RangeVal,
+                                     single_nominal_capacitance: RangeVal,
+                                     part_spec: str, footprint_spec: str) -> None:
+
+    # capacitance: user-specified capacitance
+    # single nominal capacitance: no single cap with requested capacitance, must generate multiple parallel caps
+    # nominal capacitance: selected part's capacitance
+
+    def valid_package_min_nominal_capacitance(package: int, capacitance: RangeVal, voltage: RangeVal) -> Tuple[float, float]:
+      """
+      Calculates if the package is a valid choice for the given capacitance and voltage.
+      If the package is a valid choice, the function returns the minimum nominal capacitance
+      for the specs. If the package is not valid, the function returns False.
+      """
```

Comment:
The type signature and description seem to disagree?

(looking at your code, I think you just need to update the type signature)

Also, consider returning an Optional[float] type - which can either be a float or `None`. I think that's the recommended pattern in this case.

You capacitance and voltage inputs might also be a Tuple[float, float] instead of a RangeVal.